### PR TITLE
Implementing gym leader / e4 only type theming mode, plus some ui sup…

### PR DESCRIPTION
**Pull request by VoidModer**
This is a cleaned up version of something I implemented for a customized nuzlocke run. It changes 2 main things:

1. Added a new trainer randomization mode that would apply type themes, but only for gym trainers & elite 4 members.
2. Converted the TrainersMod radio buttons to a ComboBox so stuff would fit on the page.

I've done a fair amount of bug testing across different rom versions and configurations, checking the log files for this and other modes to make sure they behaved appropriately, but I don't really know the project well enough to be confident that I hit everything important. 

Lmk if you have any questions. Thanks!

### CODE REVIEW

**Comment by VoidModer** [comment]
Here's some screenshots of the UI change that I forgot to attach in the initial post.

randomizer1
randomizer2
e

File: src/com/dabomstew/pkrandom/Settings.java [hunk 1]
```diff
@@ -187,6 +187,7 @@
     private boolean doubleBattleMode;
     private boolean shinyChance;
 
+
```
Comment (Ajarmar) [inline src/com/dabomstew/pkrandom/Settings.java:190]:
```suggestion

```

**Review by Ajarmar — CHANGES_REQUESTED** [review]
Hi, thanks for the PR (and sorry for the late response). We're definitely interested in a setting like this. There are a few things I'd like to see changed before we merge this.

- The name of the setting (both as it's presented in the GUI and in the code) is a bit misleading; there are regular trainers (and champions) in the games that have type themes, so "shuffle type themes" doesn't properly convey what the setting does. A more literal description would be better, such as "Type Themed (Gyms/Elite Four Only)"
- Some implementation details, see below

**Comment by Ajarmar** [comment]
Closing due to inactivity.

**Comment by Terra-B-Welch** [comment]
Hate to be a gravedigger, but was any progress ever made on this setting? I'd love to see this in future versions of the randomizer.

**Comment by Ajarmar** [comment]
> Hate to be a gravedigger, but was any progress ever made on this setting? I'd love to see this in future versions of the randomizer.

Yes, it was added in https://github.com/Ajarmar/universal-pokemon-randomizer-zx/commit/51db48c59f70dc6a69b7b16116e4558b0f2662f9 and will be part of the next release

**Pull request closed without merging**